### PR TITLE
Add GitLab pipeline to feature_cicd_fastlane

### PR DESCRIPTION
**Pull request by RomanovaPrime**
#### Overview

> Closes: #747

Adds option GitLab as part of CICD feature. Implements the CI part of the pipeline

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [ ] ~UI checked in Light / Dark mode~

*Stability*
- [ ] ~Checked if changes affect any features and verified affected features work as expected~
- [x] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [x] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [ ] ~Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations~
- [ ] ~If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes~

### CODE REVIEW

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}:12]: Let's use the latest flutter version

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}:55]: Please use the same exclusions as in github

**Review by StanevPrime — CHANGES_REQUESTED** [review]
Please update the
-  rx_bloc_cli documentation (README.md file)
- CHANGELOG.md
- package versions in pubspec.yaml

File: packages/rx_bloc_cli/CHANGELOG.md
Comment (StanevPrime) [inline packages/rx_bloc_cli/CHANGELOG.md:1]: Let's deliver this right after the breaking change. With this being said let's target 5.1.0

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md:158]: This is already mentioned above

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md:163]: By  environment I guess you mean OS ?

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}:49]: From what I remember `flutter pub run build_runner` is deprecated and `dart pub run build_runner` needs to be used instead. Please check this out

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}]: Done

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md:158]: I have removed it on the other place, so it will be more readable when add more options.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/docs/continuous_delivery.md]: Changed it

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_gitlab}}.gitlab-ci.yml{{/cicd_gitlab}}]: You are right - changed

File: packages/rx_bloc_cli/CHANGELOG.md
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/CHANGELOG.md]: Will do

**Review by StanevPrime — CHANGES_REQUESTED** [review]
In general looks good, but I believe we either need to wait for the CD PR before we merge that PR or need to communicate in the changelog and documentation that it includes CI only

File: packages/rx_bloc_cli/lib/src/models/generator_arguments.dart
Comment (DDavidPrime) [inline packages/rx_bloc_cli/lib/src/models/generator_arguments.dart:138]: The PR `fix/rx_bloc-score` includes changes to increase coverage of the rx_bloc_cli package. Since I see that this will be released as the `5.1.0` (meaning after the rx_bloc ecosystem gets an update), you can include those changes to your branch after it is merged And add your code-specific tests.

File: packages/rx_bloc_cli/pubspec.yaml
Comment (DDavidPrime) [inline packages/rx_bloc_cli/pubspec.yaml:3]: Don't forget to update the `rxBlocCliPackageVersion` inside the `packages/rx_bloc_cli/lib/src/rx_bloc_cli_constants.dart` file.

**Review by DDavidPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/pubspec.yaml
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/pubspec.yaml:3]: Done, thanks!

File: packages/rx_bloc_cli/lib/src/models/generator_arguments.dart
Comment (RomanovaPrime) [reply packages/rx_bloc_cli/lib/src/models/generator_arguments.dart:138]: I have updated with the latest changes, but I am not sure what kind of tests I can write for pipeline. Have you ever written such?

Comment (RomanovaPrime) [reply packages/rx_bloc_cli/lib/src/models/generator_arguments.dart:138]: I have updated tests, regarding your explanation, thanks for it! Coverage now looks suffitiont

**Review by DDavidPrime — APPROVED** [review]

**Comment by RomanovaPrime** [comment]
The implementation will stay as WIP. Currently not fully complete the ACs and moved into rx_bloc_cli_private as a brick

**Pull request closed without merging**
